src/compressed_tensors/quantization/lifecycle/initialize.py

```diff
@@ -144,7 +144,7 @@ def _initialize_scale_zero_point(
     # in the offloaded case, there's no point moving tensors to the execution device
     # if they're going to be immediately offloaded by `register_offload_parameter`
     params_device = next(module.parameters()).device
-    device = "cpu" if has_offloaded_params(module) else params_device
+    device = module._hf_hook.execution_device if has_offloaded_params(module) else params_device
```

Comment:
I dont think I follow.
iirc, we started with cpu as the param would be offloaded right after given the module is offloaded. 
Is this just an FP8 workaround?

Reply:
> Is this just an FP8 workaround? 
Yes, this change is specifically relevant to FP8

> I would expect it to be on the execution device for any ops, even if we start on the cpu
This is true. These changes make sure that initialization is also included in the list of ops that happen on quantized values

Reply:
Said another way, the act of initializing (filling with zeros) is an operation that needs to happen on the execution device